environment_annotation.yml

```diff
@@ -0,0 +1,341 @@
+  - mdurl=0.1.0
+  - mkl=2022.2.1
+  - mpc=1.3.1
+  - mpfr=4.2.0
+  - mpmath=1.3.0
+  - msgpack-python=1.0.5
+  - mypy_extensions=1.0.0
+  - mysql-common=8.0.33
+  - mysql-libs=8.0.33
+  - napari=0.4.18
+  - napari-console=0.0.8
+  - napari-plugin-engine=0.2.0
+  - napari-svg=0.1.10
+  - ncurses=6.4
+  - nest-asyncio=1.5.6
+  - networkx=3.1
+  - nodeenv=1.8.0
+  - npe2=0.7.0
+  - nspr=4.35
+  - nss=3.89
+  - numcodecs=0.11.0
+  - numexpr=2.8.4
+  - numpy=1.25.1
+  - numpydoc=1.5.0
+  - openjpeg=2.5.0
+  - openssl=3.1.2
+  - orc=1.9.0
+  - packaging=23.1
+  - pandas=2.0.3
+  - parso=0.8.3
+  - partd=1.4.0
+  - pcre2=10.40
+  - pexpect=4.8.0
+  - pickleshare=0.7.5
+  - pillow=10.0.0
+  - pint=0.22
+  - pip=23.1.2
+  - pkgutil-resolve-name=1.3.10
+  - platformdirs=3.10.0
+  - ply=3.11
+  - pooch=1.7.0
+  - pre-commit=3.3.3
+  - prompt-toolkit=3.0.39
+  - prompt_toolkit=3.0.39
+  - psutil=5.9.5
+  - psygnal=0.9.1
+  - pthread-stubs=0.4
+  - ptyprocess=0.7.0
+  - pure_eval=0.2.2
+  - py-cpuinfo=9.0.0
+  - pycparser=2.21
+  - pydantic=1.10.10
+  - pygments=2.15.1
+  - pyopengl=3.1.6
+  - pyproject_hooks=1.0.0
+  - pyqt=5.15.7
+  - pyqt5-sip=12.11.0
+  - pysocks=1.7.1
+  - pytables=3.8.0
+  - python=3.11.4
+  - python-build=0.10.0
+  - python-dateutil=2.8.2
+  - python-tzdata=2023.3
+  - python_abi=3.11
+  - pytomlpp=1.0.13
+  - pytorch=2.0.0
+  - pytz=2023.3
+  - pywavelets=1.4.1
+  - pyyaml=6.0
+  - pyzmq=25.1.0
+  - qt-main=5.15.8
+  - qtconsole-base=5.4.3
+  - qtpy=2.3.1
+  - re2=2023.03.02
+  - readline=8.2
+  - referencing=0.29.1
+  - requests=2.31.0
+  - rich=13.4.2
+  - rpds-py=0.8.10
+  - scikit-image=0.21.0
+  - scikit-learn=1.3.0
+  - scipy=1.11.1
+  - setuptools=68.0.0
+  - shellingham=1.5.1
+  - sip=6.7.9
+  - six=1.16.0
+  - sleef=3.5.1
+  - snappy=1.1.10
+  - snowballstemmer=2.2.0
+  - sortedcontainers=2.4.0
+  - sphinx=4.5.0
+  - sphinxcontrib-applehelp=1.0.4
+  - sphinxcontrib-devhelp=1.0.2
+  - sphinxcontrib-htmlhelp=2.0.1
+  - sphinxcontrib-jsmath=1.0.1
+  - sphinxcontrib-qthelp=1.0.3
+  - sphinxcontrib-serializinghtml=1.1.5
+  - stack_data=0.6.2
+  - superqt=0.4.1
+  - sympy=1.12
+  - tbb=2021.9.0
+  - tblib=1.7.0
+  - threadpoolctl=3.1.0
+  - tifffile=2023.7.10
+  - tk=8.6.12
+  - toml=0.10.2
+  - tomli=2.0.1
+  - toolz=0.12.0
+  - tornado=6.3.2
+  - tqdm=4.65.0
+  - traitlets=5.9.0
+  - typer=0.9.0
+  - typing-extensions=4.7.1
+  - typing_extensions=4.7.1
+  - tzdata=2023c
+  - ukkonen=1.0.1
+  - virtualenv=20.24.3
+  - vispy=0.12.2
+  - wcwidth=0.2.6
+  - wheel=0.40.0
+  - wrapt=1.15.0
+  - xorg-libxau=1.0.11
+  - xorg-libxdmcp=1.1.3
+  - xyzservices=2023.5.0
+  - xz=5.2.6
+  - yaml=0.2.5
+  - zarr=2.15.0
+  - zeromq=4.3.4
+  - zfp=1.0.0
+  - zict=3.0.0
+  - zipp=3.16.0
+  - zlib=1.2.13
+  - zlib-ng=2.0.7
+  - zstd=1.5.2
+  - pip:
+      - absl-py==1.4.0
+      - cachetools==5.3.1
+      - cycler==0.11.0
+      - fonttools==4.40.0
+      - google-auth==2.21.0
+      - google-auth-oauthlib==1.0.0
+      - grace==0.0.1
```

Comment:
I think the environment gets confused when instructed to install grace, which is not on `pip` just yet. 

```sh
Pip subprocess error:
ERROR: Could not find a version that satisfies the requirement grace==0.0.1 (from versions: ... )
ERROR: No matching distribution found for grace==0.0.1

failed

CondaEnvException: Pip failed
```

Reply:
This is something we need to keep changing whenever we make a new environment (because we install grace with the pip . -e, it considers grace a "pip installed" package)